src/darwin/Framework/CHIP/CHIPDeviceController.mm

```diff
@@ -118,7 +118,7 @@ - (BOOL)_isRunning
 
 - (BOOL)shutdown
 {
-    dispatch_sync(_chipWorkQueue, ^{
+    dispatch_async(_chipWorkQueue, ^{
```

Comment:
Shouldn't you re-order this and the subsequent call to `StopEventLoopTask`? Otherwise, In between the `dispatch_async` and the subsequent call to `StopEventLoopTask`, any other blocks that get dispatched to the chip work queue could end up invoking already destructed handlers? Even if we think that might not happen, I'd think it's best to actually stop the event loop from running, and then handle shutdown.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/7679 to track this.